compose/src/main/java/com/maplibre/compose/MapLibreStyleProvider.kt

```diff
@@ -0,0 +1,54 @@
+package com.maplibre.compose
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.staticCompositionLocalOf
+
+/**
+ * Create a map style provider to inject your desired light and dark mode style URLs
+ * into the Composable hierarchy.
+ *
+ * @property lightModeStyleUrl
+ * @property darkModeStyleUrl
+ */
+class MapLibreStyleProvider(
+    private var lightModeStyleUrl: String,
+    private var darkModeStyleUrl: String
+) {
+    fun getStyleUrl(isDarkMode: Boolean): String {
+        return if (isDarkMode) darkModeStyleUrl else lightModeStyleUrl
+    }
+}
+
+val LocalMapLibreStyleProvider = staticCompositionLocalOf<MapLibreStyleProvider> {
+    error("No MapLibreStyleProvider provided")
```

Comment:
I played with this a bit and think it's relatively reasonable with some better documentation (pushed). Especially since there's no good default/backup that doesn't require some of its own setup (e.g. a res/string or static/demo provider).